rust/connlib/libs/common/src/messages.rs

```diff
@@ -55,7 +55,16 @@ pub struct Peer {
     /// Peer's Ipv6 (only 1 ipv6 per peer for now and mandatory).
     pub ipv6: Ipv6Addr,
     /// Preshared key for the given peer.
-    pub preshared_key: Key,
+    pub preshared_key: SecretKey,
+}
+
+impl PartialEq for Peer {
+    fn eq(&self, other: &Self) -> bool {
+        self.persistent_keepalive.eq(&other.persistent_keepalive)
+            && self.public_key.eq(&other.public_key)
+            && self.ipv4.eq(&other.ipv4)
+            && self.ipv6.eq(&other.ipv6)
```

Comment:
I've left out comparison of the secret on purpose. These kind of comparisons are usually not constant-time which I am guessing is the reason why `Secret` doesn't implement `PartialEq`. It theoretically opens you up to timing attacks that can leak these kind of secrets if the attacker can trigger and observe the comparison.

Reply:
While it'd be difficult to exploit in this particular case, this is a great catch!